### PR TITLE
[vcpkg-docs] Add docs/users/binarycaching.md

### PR DESCRIPTION
**Pull request by ras0219**
Add documentation for binary caching, including examples for GitHub Packages and Azure DevOps Artifacts.

### CODE REVIEW

File: docs/users/binarycaching.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
+# Binarycaching
```
Comment (BillyONeal) [inline docs/users/binarycaching.md:1]:
```suggestion
# Binary Caching
```

File: docs/users/binarycaching.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+Binary caching is Vcpkg's method for reusing package builds between projects and between machines. Think of it as a "package restore accelerator" that gives you the same results as though you built from source. Each build is packaged independently, so changing one library only requires rebuilding consuming libraries.
```
Comment (BillyONeal) [inline docs/users/binarycaching.md:3]:
```suggestion
Binary caching is vcpkg's method for reusing package builds between projects and between machines. Think of it as a "package restore accelerator" that gives you the same results as though you built from source. Each build is packaged independently, so changing one library only requires rebuilding consuming libraries.
```

File: docs/users/binarycaching.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+### Valid source strings (`<source>`)
+
+| form                        | description
+|-----------------------------|---------------
+| `clear`                     | Removes all previous sources (including the default)
+| `default[,<rw>]`            | Adds the default file-based location
+| `files,<path>[,<rw>]`       | Adds a custom file-based location
+| `nuget,<uri>[,<rw>]`        | Adds a NuGet-based source; equivalent to the `-Source` parameter of the NuGet CLI
+| `nugetconfig,<path>[,<rw>]` | Adds a NuGet-config-file-based source; equivalent to the `-Config` parameter <br>of the NuGet CLI. This config should specify `defaultPushSource` for uploads.
+| `interactive`               | Enables interactive credential management for NuGet (also requires `--debug` on the command line)
```
Comment (BillyONeal) [inline docs/users/binarycaching.md:108]: If this requires `--debug`, it seems like we shouldn't document it?

File: toolsrc/src/vcpkg/binarycaching.cpp [hunk 4]
```diff
@@ -1092,8 +1092,10 @@ void vcpkg::help_topic_binary_caching(const VcpkgPaths&)
         System::print2(
             "\nBased on your system settings, the default path to store binaries is\n    ",
             fs::u8string(p_preferred.make_preferred()),
-            "\n\nThis consults %LOCALAPPDATA%/%APPDATA% on Windows and $XDG_CACHE_HOME or $HOME on other platforms.");
+            "\nThis consults %LOCALAPPDATA%/%APPDATA% on Windows and $XDG_CACHE_HOME or $HOME on other platforms.\n");
```
Comment (BillyONeal) [inline toolsrc/src/vcpkg/binarycaching.cpp:1095]: It seems like p_preferred should always be preferred and not made preferred in the middle of a logging message like this.

**Review by BillyONeal — APPROVED** [review]

File: docs/users/binarycaching.md (hunk 3 above)
Comment (ras0219) [reply docs/users/binarycaching.md]: I clarified in the docs that this is for debugging, which makes more sense about how it needs to be combined with `--debug`.

**Pull request merged**